### PR TITLE
[ISSUE #30] modify mappers and add test

### PR DESCRIPTION
**Pull request by Lambert-Rao**
Improve connection, connector, client, meta, health mappers and add their unit tests.

Fix database testing issues as noted in PR #31

### CODE REVIEW

File: eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/connection/ConnectionMapper.java [hunk 1]
```diff
@@ -34,15 +35,28 @@
...
 
+    @Select("SELECT * FROM connection WHERE cluster_id = #{clusterId} AND source_id = #{sourceId} AND source_type = #{sourceType}")
+    public List<ConnectionEntity> selectByClusterIdSourceTypeAndSourceId(ConnectionEntity connectionEntity);
+
+    @Select("SELECT * FROM connection WHERE cluster_id = #{clusterId} AND sink_id = #{sinkId} AND sink_type = #{sinkType}")
+    public List<ConnectionEntity> selectByClusterIdSinkTypeAndSinkId(ConnectionEntity connectionEntity);
+
+    @Select("SELECT * FROM connection WHERE cluster_id = #{clusterId} AND source_id = #{sourceId} AND source_type = #{sourceType} "
+        + "AND create_time > #{startTime} AND create_time < #{endTime}")
+    public List<ConnectionEntity> selectByClusterIdSourceTypeAndSourceIdAndCreateTimeRange(ConnectionEntity connectionEntity,
```
Comment (Pil0tXia) [inline eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/connection/ConnectionMapper.java:52]: Is the method name too long? Maybe you can define the method from bussiness level.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -41,7 +41,7 @@ jobs:
         image: mysql:8
```
Comment (Pil0tXia) [inline .github/workflows/ci.yml:41]: Use `mysql:8.0` to suit client version.

File: eventmesh-dashboard-console/src/test/resources/application-test.yml [hunk 3]
```diff
@@ -21,9 +21,9 @@ spring:
     type: com.alibaba.druid.pool.DruidDataSource
     druid:
       driver-class-name: com.mysql.cj.jdbc.Driver
-      url: jdbc:mysql://localhost:3306/eventmesh-dashboard-test?useSSL=false&useUnicode=true&characterEncoding=utf-8&allowMultiQueries=true&zeroDateTimeBehavior=convertToNull&allowPublicKeyRetrieval=true
+      url: jdbc:mysql://localhost:3306?useSSL=false&useUnicode=true&characterEncoding=utf-8&allowMultiQueries=true&zeroDateTimeBehavior=convertToNull&allowPublicKeyRetrieval=true
```
[the remark this answers is not in the document]
Comment (Lambert-Rao) [reply eventmesh-dashboard-console/src/test/resources/application-test.yml:24]: in testing progress, I init database like this:
```sql
CREATE DATABASE IF NOT EXISTS `eventmesh_dashboard_test`;
USE `eventmesh_dashboard_test`
DROP TABLE IF EXISTS `connection`;
CREATE TABLE `connection`
```
I wonder if I add db name here will it affect the testing process

File: eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/connection/ConnectionMapper.java (hunk 1 above)
Comment (Lambert-Rao) [reply eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/connection/ConnectionMapper.java:52]: I can't find a suitable name, and this method will be encapsulated in the service code, so I think this is OK.

File: eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/health/HealthCheckResultMapper.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    @Select({
+        "<script>",
+        "   SELECT * FROM health_check_result",
+        "   WHERE (cluster_id, type, type_id, status) IN",
+        "   <foreach collection='list' item='item' open='(' separator=',' close=')'>",
+        "       (#{item.clusterId}, #{item.type}, #{item.typeId}, 2)",
+        "   </foreach>",
+        "   ORDER BY create_time DESC",
+        "</script>"
+    })
```
[the remark this answers is not in the document]
Comment (Lambert-Rao) [reply eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/health/HealthCheckResultMapper.java:86]: I don't know what you mean, I always use `({text,text})` format

File: eventmesh-dashboard-console/src/test/resources/application-test.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Pil0tXia) [reply eventmesh-dashboard-console/src/test/resources/application-test.yml:24]: You may have I try and better unified.

File: eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/connection/ConnectionMapper.java (hunk 1 above)
Comment (Pil0tXia) [reply eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/connection/ConnectionMapper.java:52]: alright

File: eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/health/HealthCheckResultMapper.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (Pil0tXia) [reply eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/health/HealthCheckResultMapper.java:86]: like this:

image

[the remark this answers is not in the document]
Comment (Lambert-Rao) [reply eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/health/HealthCheckResultMapper.java:86]: I would use array format when using dynamic scripts (`<script>`), and `+` format when using simple statements.

File: eventmesh-dashboard-console/src/test/resources/application-test.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Lambert-Rao) [reply eventmesh-dashboard-console/src/test/resources/application-test.yml:24]: will keep the db name

File: eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/health/HealthCheckResultMapper.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (Pil0tXia) [reply eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/health/HealthCheckResultMapper.java:88]: ok

File: eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/connection/ConnectionMapper.java [hunk 5]
```diff
@@ -58,34 +58,34 @@ public List<ConnectionEntity> selectByClusterIdSinkTypeAndSinkIdAndCreateTimeRan
         @Param("startTime") Timestamp startTime, @Param("endTime") Timestamp endTime);
 
     @Options(useGeneratedKeys = true, keyProperty = "id", keyColumn = "id")
-    @Insert("INSERT INTO connection (cluster_id, source_type, source_id,"
-        + " sink_type, sink_id, runtime_id, status, topic, group_id, description)"
+    @Insert("INSERT INTO connection (cluster_id, source_type, source_id," + " sink_type, sink_id, runtime_id, status, topic, group_id, description)"
         + " VALUES ( #{clusterId}, #{sourceType}, #{sourceId}, "
```
Comment (Pil0tXia) [inline eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/connection/ConnectionMapper.java:62]: line break removed?

Comment (Lambert-Rao) [reply eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/connection/ConnectionMapper.java:62]: mistakenly do this, I will fix it in future commits

Comment (Pil0tXia) [reply eventmesh-dashboard-console/src/main/java/org/apache/eventmesh/dashboard/console/mapper/connection/ConnectionMapper.java:62]: ok

**Review by Pil0tXia — APPROVED** [review]

**Pull request merged**